ledokku-bootstrap.sh

```diff
@@ -1,39 +1,162 @@
+  fi
+}
+
+function check-root() {
+  
+  #Check root and if not root take permissions (Some providers doesnot support passwordless sudo)
+  ## It is always better to do this.
+  ## We will not face any further issues, during any sort of compulsory sudo commands; like the case for installing plugins in Dokku or Giving permissions to our scripts
+  if [ "$(whoami)" == "root" ] ; then
+      echo "${YELLOW}Nice you are running the script as root!${END}"
+  else
+      echo "${RED}Please! Run the script with root access${END}, without root access I cannot create plugins in Dokku"
+      exit
+  fi
+}
+
+function system-update {
+
+  # Updating and Upgrading system
+  ## If dokku was installed, it will be automatically updated to the latest version
+  ## Staying up to date is always good
+  if (whiptail --title "Update and Upgrade System " --yes-button "Yes" --no-button "Skip"  --yesno "Do you wish to Update packges and Upgrade your system?" 10 60) then
+      echo "You chose to update your system"
+      # Update and skip to next step
+      echo "${YELLOW}Updating System${END}"
+      sudo dpkg --configure -a
+      sudo apt -y --purge autoremove
+      sudo apt install -f
+      sudo apt -y update
+      wait
+      echo "${YELLOW}Upgrading System${END}"
+      sudo apt -y upgrade
+      sudo apt -y autoclean
+      sudo apt -y --purge autoremove &
+      process_id=$!
+      wait $process_id
+      echo "Exit status: $?"
+  else
+      echo "${YELLOW}You chose to skip.${END}"
+      # Should skip to next step
+  fi
+}
+
+function ensure-dokku() {
+  
+  # Confirming the existance of Dokku
+  ## If exists, promt for update
+  ## If not exits Let him download.
+  if which dokku >/dev/null; then
+      echo "${GREEN}Dokku Exists${END}"
+      # Promt for update
+      if (whiptail --title "Updating Dokku" --yes-button "Update" --no-button "Skip"  --yesno "Would you like to update your Dokku?" 10 60) then
+          echo "${YELLOW}You chose Update.${END}"
+          # Update Dokku
+          echo "${YELLOW}Upgrading Dokku${END}"
+          sudo apt-get -y update -qq
+          wait
+          sudo apt-get -qq -y --no-install-recommends install dokku herokuish sshcommand plugn gliderlabs-sigil dokku-update dokku-event-listener
+          wait
+          sudo apt -y upgrade &
+          process_id=$!
+          wait $process_id
+          echo "Exit status: $?"
+          echo "${YELLOW}Now you have the latest Version of Dokku${END}"
+          # Dokku Updated
+      else
+          echo "${Yellow}You chose to skip dokku updates.${END}"
+          # Dokku Update skipped
+      fi
+  else
+      echo "${RED}Dokku does not exist${END}"
+      # Show messagebox and make it mandatory to download and install dokku
+      whiptail --title "Unable to Detect Dokku" --msgbox "If you want to insatll your app using t2d, it is madatory to install Dokku. So, I would like to install Dokku on behalf of you." 10 60
```

Comment:
What is t2d?
Also typo on `insatll`
Offering to install dokku is a great idea!

Reply:
**Edit 02**: I will clear these typos.

t2d => Ledokku
insatll => Install